staging/src/k8s.io/kubectl/pkg/util/podutils/podutils.go

```diff
@@ -190,14 +199,32 @@ func podReadyTime(pod *corev1.Pod) *metav1.Time {
 	return &metav1.Time{}
 }
 
-func maxContainerRestarts(pod *corev1.Pod) int {
+func maxContainerRestarts(cs []corev1.ContainerStatus) int {
```

Comment:
Why don't we get the pod here and take restartable init containers into account to remove the redundancy?

Reply:
Sorry for the confusion.
I meant one maximum count across all containers including regular containers and sidecar containers.
What do you think?

Reply:
This is inconsistent with what we originally designed.

> I think the desireable behavior will be to check regular containers max restart count first. And compare this.
Then compare max restart count for restarteable init containers.

/cc @SergeyKanzhelev

Reply:
> This is inconsistent with what we originally designed.
> 
> > I think the desireable behavior will be to check regular containers max restart count first. And compare this.
> > Then compare max restart count for restarteable init containers.

Sounds reasonable.